lib/components/base/EnvironmentIndicator.vue

```diff
@@ -47,52 +47,29 @@
     </template>
   </span>
 </template>
-<script setup>
+<script setup lang="ts">
 import { GlobeIcon, ClientIcon, ServerIcon, InfoIcon } from '@'
-</script>
-<script>
-import { defineComponent } from 'vue'
 
-export default defineComponent({
-  props: {
-    type: {
-      type: String,
-      default: 'mod',
-    },
-    serverSide: {
-      type: String,
-      required: false,
-      default: '',
-    },
-    clientSide: {
-      type: String,
-      required: false,
-      default: '',
-    },
-    typeOnly: {
-      type: Boolean,
-      required: false,
-      default: false,
-    },
-    alwaysShow: {
-      type: Boolean,
-      required: false,
-      default: false,
-    },
-    search: {
-      type: Boolean,
-      required: false,
-      default: false,
-    },
-    categories: {
-      type: Array,
-      required: false,
-      default() {
-        return []
-      },
-    },
-  },
-})
+withDefaults(
+  defineProps<{
+    type: string
+    serverSide?: string
+    clientSide?: string
+    typeOnly?: boolean
+    alwaysShow?: boolean
+    search?: boolean
+    categories?: Array<string>
```

Comment:
```suggestion
    categories?: string[]
```

Use shorter syntax for array types.

Reply:
Done ^^.